### PR TITLE
Feature/add scm submodule

### PR DESCRIPTION
**Pull request by Minimonium**
Implements #3020

Allows to set additional SCM attribute `"submodule":["shallow", "recursive"]`. 
Default behaviour is the same as before - no submodule support.
Changed `normal` attribute value to `shallow` since it have more meaning.
1. Is `submodule` a good name for an attribute? Would it be misleading?
2. Should there be an error on invalid `submodule` value? 
3. What did I miss? 😄

### CODE REVIEW

File: conans/client/tools/scm.py [hunk 1]
```diff
@@ -63,6 +63,14 @@ def clone(self, url, branch=None):
             branch_cmd = "--branch %s" % branch if branch else ""
             output = self.run('clone "%s" . %s' % (url, branch_cmd))
             output += self._configure_ssl_verify()
+
+        if submodule is "shallow":
```
Comment (memsharded) [inline conans/client/tools/scm.py]: You should use ``submodule == "shallow"``. Your current imlementation relies on short string optimizations, but is platform and implementation dependent.

File: conans/test/functional/scm_test.py [hunk 2]
```diff
@@ -168,6 +168,65 @@ def test_install_checked_out(self):
...
+            submodule_path = os.path.join(
+                folder, 
+                os.path.basename(os.path.normpath(submodule)))
+            subsubmodule_path = os.path.join(
+                submodule_path, 
+                os.path.basename(os.path.normpath(subsubmodule)))
+            return submodule_path, subsubmodule_path
+
+        # Check old (default) behaviour
+        conanfile = tmp.format(url=path, revision=commit, submodule="none")
```
Comment (memsharded) [inline conans/test/functional/scm_test.py]: I think this ``"none"`` evidence a possible source of user UI errors. I would suggest:
- Having not defined values "shallow, recursive", is an error and will raise an Exception. This avoids users figuring out why it doesnt work if they do a typo or something.
- This line here will skip the "submodule" field completely, not use a "None" value.

File: conans/client/tools/scm.py (hunk 1 above)
Comment (Minimonium) [reply conans/client/tools/scm.py]: Good to know! Thanks.

File: conans/test/functional/scm_test.py (hunk 2 above)
Comment (Minimonium) [reply conans/test/functional/scm_test.py]: Yeah, I think I derped here a bit. "none" here was meaning just random value, not a "shallow, recursive" which is strange behavior. I will solidify that part later! Thanks.

**Comment by memsharded** [comment]
This has been merged in commits in https://github.com/conan-io/conan/pull/3114

**Pull request closed without merging**